Tests/Unit/ZoomApiServiceTest.php

```diff
@@ -0,0 +1,255 @@
+
+        $service = $this->getService();
+        $service->initializeObject();
+        $this->cacheMock->method('get')->willReturn(false);
+
+
+        $service->getRecordings(new DateTime('2023-01-02'), new DateTime('2023-01-01'));
+    }
+
+    /** @test */
+    public function upcomingMeetingsReturnsValidCachedData()
+    {
+        $this->cacheMock = $this->getMockBuilder(VariableFrontend::class)->disableOriginalConstructor()->getMock();
+        $this->cacheMock->expects($this->once())
+            ->method('get')
+            ->willReturn(['cached123']);
+
+        $service = $this->getService();
+        $service->initializeObject();
+
+
+        $result = $service->getUpcomingMeetings();
+
+        $this->assertEquals(['cached123'], $result);
+    }
+
+    /** @test */
+    public function upcomingMeetingsWithEmptyCacheReturnsNewData()
+    {
+        $handlerStack = HandlerStack::create(
+            new MockHandler([
+                new Response(200, [], json_encode(['meetings' => ['meeting1'], 'next_page_token' => '1234567890'])),
+                new Response(200, [], json_encode(['meetings' => ['meeting2'], 'next_page_token' => ''])),
+            ])
+        );
+
+        $service = $this->getService($handlerStack);
+        $service->initializeObject();
+        $this->cacheMock->method('get')->willReturn(false);
+
+
+        $result = $service->getUpcomingMeetings(true);
+
+
+        $this->assertEquals(['meeting1', 'meeting2'], $result);
+    }
+
+    /** @test */
+    public function upcomingMeetingsWithCacheExceptionReturnsNewData()
+    {
+        $handlerStack = HandlerStack::create(
+            new MockHandler([
+                new Response(200, [], json_encode(['meetings' => ['meeting1'], 'next_page_token' => '1234567890'])),
+                new Response(200, [], json_encode(['meetings' => ['meeting2'], 'next_page_token' => ''])),
+            ])
+        );
+
+        $service = $this->getService($handlerStack);
+        $service->initializeObject();
+        $this->cacheMock->method('get')->willReturn(false);
+        $this->cacheMock->method('set')->willThrowException(new Exception());
+        $this->systemLoggerMock->shouldReceive('error')->once();
+
+
+        $result = $service->getUpcomingMeetings(true);
+
+
+        $this->assertEquals(['meeting1', 'meeting2'], $result);
+    }
+
+    /** @test */
+    public function getUpcomingMeetingsWithEmptyResponseThrowsException()
+    {
+        $this->expectException(ZoomApiException::class);
+
+        $handlerStack = HandlerStack::create(
+            new MockHandler([
+                new Response(200, [], ''),
+            ])
+        );
+
+        $service = $this->getService($handlerStack);
+        $service->initializeObject();
+        $this->cacheMock->method('get')->willReturn(false);
+
+
+        $service->getUpcomingMeetings(true);
+    }
+
+    /**
+     * @param Client $client
```

Comment:
This is wrong now.
One of the many cases where a redundant DocBlock comment makes things worse instead of better :D